lib/rubocop/cop/lint/literal_as_condition.rb

```diff
@@ -98,13 +99,43 @@ def message(node)
 
         def check_for_literal(node)
           cond = condition(node)
-          if cond.literal?
-            add_offense(cond)
+          if cond.truthy_literal?
+            autocorrect_redundant_truthy_condition(node, cond)
+          elsif cond.falsey_literal?
+            autocorrect_redundant_falsey_condition(node, cond)
           else
             check_node(cond)
           end
         end
 
+        def autocorrect_redundant_truthy_condition(node, cond)
```

Comment:
Splitting the autocorrection into `autocorrect_redundant_truthy_condition` and `autocorrect_redundant_falsey_condition` means there's some duplicate code shared between them, but unifying them triggers a lot of metrics and line length failures.
```ruby
def autocorrect_redundant_condition(node, cond)
  if (cond.truthy_literal? && node.if_type?) || (cond.falsey_literal? && node.if_type? && node.unless?)
    add_offense(cond) do |corrector|
      corrector.replace(node, node.if_branch.source)
    end
  elsif (cond.truthy_literal? && (node.while_type? || node.while_post_type?)) || (cond.falsey_literal? && (node.until_type? || node.until_post_type?))
    add_offense(cond) do |corrector|
      corrector.replace(cond, cond.truthy_literal?.to_s)
    end
  else
    add_offense(cond)
  end
end
```